extensions/http/jersey/src/main/java/org/eclipse/dataspaceconnector/extension/jersey/validation/ValidationFunctionRegistryImpl.java

```diff
@@ -0,0 +1,42 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.extension.jersey.validation;
+
+import java.lang.reflect.Method;
+
+public class ValidationFunctionRegistryImpl implements ValidationFunctionRegistry {
```

Comment:
Seems like an avoidable indirection layer, could `ResourceInterceptorProvider` directly implement `ValidationFunctionRegistry`?

Reply:
technically we could do that, but the idea was to make the `ResourceInterceptorProvider` a generic hook point that could be used for other things than _just_ validation, so it is a generic layer, whereas the `ValidationFunctionRegistry` "binds" it specifically to the validation use case. 

Also, I liked the idea of having a clean user-facing interface.

Reply:
hmmm, so I gave it a try, and I actually like it. check the respective [commit](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/pull/1756/commits/4b835e8bfffa994db6a2e1510b2a6b6bcfe5748b)

Reply:
@paullatzelsperger maybe the `ValidationFunctionRegistry` could be called `InterceptorFunctionRegistry`?
That would make it more generic looking.

Reply:
I thought about it, and considering that everything else is called "interceptor"-something I'm fine with it.